lib/ansible/plugins/connection/network_cli.py

```diff
@@ -181,10 +181,10 @@ def _handle_prompt(self, resp, obj):
         """Matches the command prompt and responds"""
         if isinstance(obj, (binary_type, text_type)) or not isinstance(obj['prompt'], Sequence):
             obj['prompt'] = [obj['prompt']]
-        prompts = [re.compile(r, re.I) for r in obj['prompt']]
+        prompts = [re.compile(to_text(r, errors='surrogate_or_strict'), re.I) for r in obj['prompt']]
         answer = obj['answer']
         for regex in prompts:
-            match = regex.search(resp)
+            match = regex.search(to_text(resp, errors='surrogate_or_strict'))
```

Comment:
Same here.  Since both resp and obj['prompt'] are supposed to contain byte strings we need to continue that.